vcr/issuer/statuslist2021_store.go

```diff
@@ -0,0 +1,221 @@
+package issuer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/vcr/types"
+	"strconv"
+	"sync"
+
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+	"github.com/nuts-foundation/nuts-node/vcr/credential/statuslist2021"
+	"github.com/nuts-foundation/nuts-node/vdr/didweb"
+)
+
+// errNotFound wraps types.ErrNotFound to clarify which credential is not found
+var errNotFound = fmt.Errorf("status list: %w", types.ErrNotFound)
+
+// errUnsupportedPurpose limits current usage to 'revocation'
+var errUnsupportedPurpose = errors.New("status list: purpose not supported")
+
+// errNotFound wraps types.ErrRevoked to clarify the source of the error
+var errRevoked = fmt.Errorf("status list: %w", types.ErrRevoked)
+
+type StatusPurpose string
+
+const (
+	StatusPurposeRevocation = "revocation"
+	statusPurposeSuspension = "suspension" // currently not supported
+)
+
+// StatusList2021Store keeps track of the number of credentials with a credential status per issuer, and allows revoking
+// them by setting the relevant bit on the StatusList.
+type StatusList2021Store interface {
```

Comment:
could this type & friends (e.g. `StatusList2021CredentialSubject`) be bundled in a `statuslist2021` package?

Reply:
will be done in a next PR that does not make any functional changes